keps/sig-node/4176-cpumanager-spread-cpus-preferred-policy/README.md

```diff
@@ -0,0 +1,824 @@
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerPolicyAlphaOptions`
+  - Components depending on the feature gate: `kubelet`
+- [x] Change the kubelet configuration to set a CPUManager policy of static and a CPUManager policy option of `spread-physical-cpus-preferred`
+  - Will enabling / disabling the feature require downtime of the control plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning of a node? (Do not assume Dynamic Kubelet Config feature is enabled). Yes -- a kubelet restart is required.
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+No. In order to enable the feature, the user must explicitly set the `CPUManager` policy to static, enable `CPUManagerPolicyAlphaOptions` and the CPUManager policy option to `spread-physical-cpus-preferred`.
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+For sure. The feature gate can be disabled in many ways.
+- Set `CPUManager` policy to `none`
+- Disable `CPUManagerPolicyAlphaOptions`
+- select other policy options instead of `spread-physical-cpus-preferred`
+
+Ongoing workloads will maintain their operations without disruption, while upcoming tasks will receive CPU allocations in accordance with the reinstated policy.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If we reactivate the feature after a rollback, the outcome remains unchanged. Current containers will retain their allocations, while newly created containers will be affected.
+
+###### Are there any tests for feature enablement/disablement?
+
+A dedicated e2e test will validate the preservation of the default behavior when the feature gate is turned off or when the feature is unused. This will be conducted through two distinct test scenarios.
```

Comment:
Also test when policy option is set to `spread-physical-cpus-preferred` but `CPUManagerPolicyAlphaOptions` is disabled?

Reply:
good point. added.